### PR TITLE
docs(toh): Replaced window.history with location service

### PR DESCRIPTION
**Pull request by brandonroberts**
Also updated references of "Component Router" to "Router"

### CODE REVIEW

File: public/docs/_examples/toh-5/ts/app/hero-detail.component.ts [hunk 1]
```diff
@@ -38,7 +40,7 @@ export class HeroDetailComponent implements OnInit {
 
   // #docregion goBack
   goBack(): void {
-    window.history.back();
+    this.location.back();
```
Comment (chalin) [inline public/docs/_examples/toh-5/ts/app/hero-detail.component.ts]: Nice.

File: public/docs/_examples/toh-6/ts/app/hero-detail.component.ts [hunk 2]
```diff
@@ -29,11 +31,11 @@ export class HeroDetailComponent implements OnInit {
   // #docregion save
   save(): void {
     this.heroService.update(this.hero)
-      .then(this.goBack);
+      .then(() => this.goBack());
```
Comment (chalin) [inline public/docs/_examples/toh-6/ts/app/hero-detail.component.ts]: Why this change? You are replacing a function g by an anonymous function which does nothing but call g. I.e., g and `() => g()` are equivalent, though the former is shorter.

Comment (brandonroberts) [inline public/docs/_examples/toh-6/ts/app/hero-detail.component.ts]: I was getting an error before that it `Cannot call location of null` error. Maybe it was a fluke on my part but I just tried it again and it worked correctly. I'll change it back to what it was.

Comment (chalin) [inline public/docs/_examples/toh-6/ts/app/hero-detail.component.ts]: Oops, your change is actually the right thing to do to ensure that `this` gets bound properly in `goBack`.
